crates/core/src/config.rs

```diff
@@ -93,8 +122,39 @@ pub struct LogConfig {
     pub directives: Vec<String>,
 }
 
-impl ConfigFile {
-    pub fn read(path: &ConfigToml) -> anyhow::Result<Option<Self>> {
-        parse_config(path.as_ref())
+/// Patch the value of a key in a TOML document,
+/// preserving the formatting and comments of the original value.
+fn patch_value(item: Option<&Item>, value: Option<&str>) -> Option<Item> {
```

Comment:
nit: maybe we could call this something like `update_value_preserve_decor` or something? I didn't really understand the point of `patch_value` without reading through the cases.